src/textual.rs

```diff
@@ -52,6 +52,7 @@ pub use self::{parser::Parser, pretty::Prettifier};
 pub struct NamedUniverse {
     names: HashMap<String, Sym>,
     syms: HashMap<Sym, String>,
+    vars: Vec<String>,
```

Comment:
Storing the mapping between variable names and indices in the universe is not ideal, because then all variables share a single scope.
The query engine is written in a way that assumes that each rule has its own local scope. Things will still work this way, but rules will allocate a lot of unbound and unused variables during instantiation this way:

https://github.com/fatho/logru/blob/280d6cb6a591d632a2d33066a6b4c7297459a51e/src/ast.rs#L82-L92
https://github.com/fatho/logru/blob/280d6cb6a591d632a2d33066a6b4c7297459a51e/src/solver.rs#L469